packages/labs/ssr/src/lib/render-value.ts

```diff
@@ -223,6 +223,14 @@ type Op =
   | CustomElementClosedOp
   | PossibleNodeMarkerOp;
 
+/**
+ * Any of these top level tags will be removed by parse5's `parseFragment` and
+ * will cause part errors if there are any part bindings. For only these tags,
+ * we use a page-level template `parse`.
+ */
+const REGEXP_TEMPLATE_STARTS_WITH_PAGE_TAG =
+  /^\s*(<(!doctype|html|head|body))/i;
```

Comment:
I think this will handle leading comments too:
```suggestion
  /^(\s*|<!--.*-->)*(<(!doctype|html|head|body))/is;
```

Reply:
Great suggestion. I've updated regex and added a bunch of tests. Things look pretty great with the comment regex addition.